client/python/README.md

```diff
@@ -9,31 +9,107 @@ pip install djclient
 ## Examples
 
 To initialize the client:
-```
-from djclient import DJClient
+```python
+from djclient.dj import DJClient, Source, Dimension, Transform, Metric
 client = DJClient("http://dj-endpoint:8000")
 ```
 
+### Catalogs and Engines
+
 To list available catalogs for the DJ host:
-```
+```python
 client.catalogs()
 ```
 
 To list available engines for the DJ host:
-```
+```python
 client.engines()
 ```
 
-ALl nodes can be found with:
+To create a catalog:
+```python
+catalog = Catalog(
+    name="prod"
+)
+catalog.publish()
+```
+
+To create an engine:
+```python
+engine = Engine(
+    name="spark",
+    version="3.2.2",
+    uri="..."
+)
+engine.publish()
```

Comment:
Is the reason for the `REGISTRY` map in order to support this `.publish()` syntax that can find the client in the global namespace? I know this is mainly opinion here but I'd find it a bit clearer to use the dj client directly, something like `dj.publish(engine)`. Plus it's only 2 more characters. :D

Reply:
Pretty much. I think some folks might prefer `engine.publish()`, but I also don't think it's a huge hassle to swap to `dj.publish(engine)`. Let me make this switch

Reply:
@samredai Although what about things like `catalog.add_engine(...)`? I think that's cleaner than `dj.add_engines_to_catalog(...)`, but it fits with the current pattern better of `<obj>.<verb>` rather than `<client>.<verb>(<ob>)`

Reply:
I see, yeah I agree that's not ideal. And having to pass the client as an argument to these object methods is only slightly better. Maybe we should stick with the original setup here then. @betodealmeida @agorajek @CircArgs any thoughts?

Reply:
Also just to clarify, by "passing the client as an argument" I meant `catalog.add_engine(dj, engine)`. Always having to pass in the client is something that's easy to forget though...

Another alternative I can think of is that new objects are created by methods on the client and the client sets itself as an attribute on the object.
```py
from djclient import DJClient

dj = DJClient("http://dj-endpoint:8000")

catalog = dj.new_catalog(
    name="prod"
)
catalog.publish()

engine = dj.new_engine(
    name="spark",
    version="3.2.2",
    uri="..."
)
engine.publish()

catalog.add_engine(engine)
```

Reply:
Another thing for catalogs: I think in the `add_catalog` endpoint, the DJ core service should also call DJQS and add the catalogs + engines there. I was considering wrapping this functionality into the client, but I think having DJ call DJQS is cleaner since then the client wouldn't need to provide/know about a DJQS endpoint.

Reply:
Having to add it in both places does seem redundant but I don't think it's too bad since these are more server administrative tasks right? I'm wondering now if we should even allow the user-facing client to create catalogs and engines. I'd expect that's configured once and rarely updated.

Reply:
Yeah, perhaps we can just remove these from the user-facing client, since I'm not sure users will create catalogs correctly anyway 😆 But I'd still prefer it if DJ core could keep DJQS in sync instead of making the server admin make both calls.

Reply:
Agree that managing catalogs and engines can be less user friendly for the sake of keeping things simple. I don't have much preferences around `object.publish()` over `dj.publish(object)`.